internal/secrets/drivertest/drivertest.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package drivertest provides a conformance test for implementations of
+// driver.
+package drivertest
+
+import (
+	"context"
+	"github.com/google/go-cloud/internal/secrets/driver"
+	"github.com/google/go-cmp/cmp"
+	"testing"
+)
+
+// Harness descibes the functionality test harnesses must provide to run
+// conformance tests.
+type Harness interface {
+	// MakeDriver returns an Encypter and a Decrypter for use in tests.
+	MakeDriver(ctx context.Context) (driver.Encrypter, driver.Decrypter, error)
```

Comment:
I suspect you will want a `Close` function on this interface as well, once you try to use it for Cloud providers (`MakeDriver` for them will likely create some kind of client that needs to be closed). But fine to add later.

Reply:
Had one in here, and realized it wouldn't be doing anything for the local package. Shall I add it anyway, since we know there *will* be one? Or should I let that be handled at the point at which it actually becomes necessary?

Reply:
Fine to let it be added later when it's needed, this was mostly an FYI.